content/en/docs/How To/create_input.md

```diff
@@ -0,0 +1,102 @@
+with read data has two fields: `Bytes` that must contain the raw read bytes (possible contains
+more records) and `Meta`.
+
+[`Metadata`](https://pkg.go.dev/github.com/AdRoll/baker#Metadata) can contain any information
+that the input wants to share with the other components of the pipeline about the chunk of bytes
+produced.  
+Typical information could be the timestamp of the read, details about the source, etc. 
+
+Those metadata will be associated to each record extracted from `Bytes` and will be available
+to the whole pipeline.
+
+## The Input interface
+
+```go
+type Input interface {
+	Run(output chan<- *Data) error
+	Stop()
+	Stats() InputStats
+	FreeMem(data *Data)
+}
+```
+
+The [Input interface](https://pkg.go.dev/github.com/AdRoll/baker#Input) must be implemented when
+creating a new input component.
+
+The `Run` function implements the component logic and receives a channel where it sends the
+[raw data](https://pkg.go.dev/github.com/AdRoll/baker#Data) it processes.
+
+`FreeMem` is the function called by Baker when a chunk of data has been completely processed and
+its memory can be used again. It's up to the input to choose whether to recycle it or not.  
+An often used pattern is to manage the objects with [`sync.Pool`](https://golang.org/pkg/sync/#Pool),
+see the [TCP](https://github.com/AdRoll/baker/blob/main/input/tcp.go),
+[List](https://github.com/AdRoll/baker/blob/main/input/list.go) or
+[SQS](https://github.com/AdRoll/baker/blob/main/input/sqs.go) inputs for inspiration.
+
+## InputDesc
+
+```go
+var MyInputDesc = baker.InputDesc{
+	Name:   "MyInput",
+	New:    NewMyInput,
+	Config: &MyInputConfig{},
+	Help:   "Help message for the input",
+}
+```
+
+This object has a `Name`, that is used in the Baker configuration file to identify the input,
+a costructor function (`New`), a config object (used to parse the input configuration in the
+TOML file) and a help text that must help the users to use the component and its configuration
+parameters.
+
+### Input constructor
+
+The `New` key in the `InputDesc` object represents the constructor function.
+
+The function receives a [InputParams](https://pkg.go.dev/github.com/AdRoll/baker#InputParams)
+object and returns an instance of [Input](https://pkg.go.dev/github.com/AdRoll/baker#Input).
+
+The function should verify the configuration params into `InputParams.DecodedConfig` and initialize
+the component. Additional operations can be performed in the `Run` function when called.
+
+### The input configuration and help
+
+The input configuration object (`MyInputConfig` in the previous example) must export all
+configuration parameters that the user can set in the TOML topology file.
+
+Each key in the struct must include a `help` string tag and a `required` boolean tag, both are
+mandatory.
+
+The former helps the user to understand the possible values of the field, the latter tells Baker
+whether to refuse a missing configuration param.
```

Comment:
:bulb:  **idea** :bulb: 
We should probably list somewhere the types we support for the fields of the Config structures, since we don't support every type there is.

Rather than describing those in each component page, we could have a page dedicated to the Config structs, in which we also list the supported struct tags (`required`, `help` and `default`).

And list this new page here and there.

What do you think?

Reply:
sounds good, we can also simplify all these components page. I'll leave the page as it is now, when we'll create the suggested page we can rework the components pages removing duplicated info